tests/handlers/test_query_results.py

```diff
@@ -152,6 +152,29 @@ def test_signed_in_user_and_different_query_result(self):
         rv = self.make_request('get', '/api/queries/{}/results/{}.json'.format(query.id, query_result2.id))
         self.assertEquals(rv.status_code, 403)
 
+    def test_query_result_query(self):
+        restricted_group = self.factory.create_group(org=self.factory.org, type=Group.REGULAR_GROUP, name="restricted")
+        db.session.commit()
```

Comment:
Why is a commit here needed and not in the other test method?

Reply:
When you need a numeric ID directly you have to commit. Instead of having a proper relation for users and groups there's a `group_ids` array column on User.